app/gui2/shared/ast/text.ts

```diff
@@ -28,52 +28,34 @@ function escapeAsCharCodes(str: string): string {
   return out
 }
 
-const escapeRegex = new RegExp(
-  `${escapeSequences.map(([_, raw]) => escapeAsCharCodes(raw)).join('|')}`,
-  'gu',
-)
+const fixedEscapes = escapeSequences.map(([_, raw]) => escapeAsCharCodes(raw))
 
-const unescapeRegex = new RegExp(
-  '\\\\(?:' +
-    `${escapeSequences.map(([escape]) => escapeAsCharCodes(escape)).join('|')}` +
-    '|x[0-9a-fA-F]{0,2}' +
-    '|u\\{[0-9a-fA-F]{0,4}\\}?' + // Lexer allows trailing } to be missing.
-    '|u[0-9a-fA-F]{0,4}' +
-    '|U[0-9a-fA-F]{0,8}' +
-    ')',
-  'gu',
-)
+const escapeRegex = new RegExp(`${[...fixedEscapes, '\\p{Surrogate}'].join('|')}`, 'gu')
```

Comment:
1. Why is it inside ``? join should return string already.
2. What is this `{Surrogate}` tag?

Reply:
Added a comment:

> Unpaired-surrogate codepoints are not technically valid in Unicode, but they are allowed in Javascript strings.
> Enso source files must be strictly UTF-8 conformant.